### DISCUSSION TITLE
使用NewFilter日志不会输出该行的行号，怎么解决??

### DISCUSSION THREAD

**Discussion by Sejenew** [go-kratos/kratos#3423] [Q&A] [answered]
logs := log.NewHelper(
		log.NewFilter(
			logger,
			log.FilterLevel(log.LevelWarn),
		),
	)
logs.info("1111111")// 调用这个后，日志不会输出该行的“调用”行号了。
// 备注：不用过滤，日志信息中有对应的行号caller=service/greeter.go:43

**Comment by Waxion** [comment, marked as answer]
如果你使用了DefaultCaller = Caller(4)来设置输出调用行号的话,可以试试改用 log.Caller(4+1),我理解是用了New** 就多套了一层 Call中的层级就要加1
